src/client/mod.rs

```diff
@@ -657,6 +660,8 @@ impl IotHubClientBuilder {
     }
 }
 
+static TRACE_ID: AtomicU32 = AtomicU32::new(0);
```

Comment:
nit: why global variable and not attribute of the twin? If we plan to use the trace_id for further testing/tracing across services, this would probably simplify testing.

Reply:
done